File: src/main/java/ModLoader.java
import net.minecraftforge.fml.common.Mod;
import rbq.wtf.lycoris.client.Client;

import javax.swing.*;

@Mod(modid = "Lycoris Client")
public class ModLoader {

    public ModLoader() {
        new Thread(() -> {
            if (Client.developEnv)
                JOptionPane.showConfirmDialog(null, "Wait");
            Client.INSTANCE.start();
        }).start();
    }
}

File: src/main/java/ModLoader.kt
import net.minecraftforge.fml.common.Mod
import rbq.lycoris.client.Client
import javax.swing.JOptionPane

@Mod(modid = "Lycoris Client")
class ModLoader {
    init {
        Thread {
            JOptionPane.showConfirmDialog(null, "Please wait for minecraft launch...")
            Client.start()
        }.start()
    }
}